packages/compass-schema-validation/src/components/validation-editor/validation-editor.jsx

```diff
@@ -97,48 +125,12 @@ class ValidationEditor extends Component {
    */
   constructor(props) {
     super(props);
-    this.completer = new ValidationAutoCompleter(
-      props.serverVersion,
-      EditorTextCompleter,
-      props.fields
-    );
     this.debounceValidatorChanged = debounce((validator) => {
       this.props.clearSampleDocuments();
       this.trackValidator(validator);
     }, 750);
   }
 
-  /**
-   * Should the component update?
-   *
-   * @param {Object} nextProps - The next properties.
-   *
-   * @returns {Boolean} If the component should update.
-   */
-  shouldComponentUpdate(nextProps) {
```

Comment:
This seems unnecessary and causes bugs because some props were not included, like dark mode. There was no explanation why this was here. I removed it, but tell me if you know the reason for it